src/librustdoc/html/render/print_item.rs

```diff
@@ -2267,6 +2271,53 @@ fn document_non_exhaustive<'a>(item: &'a clean::Item) -> impl fmt::Display + 'a
     })
 }
 
+fn document_default_header(
+    cx: &mut Context<'_>,
+    it: &clean::Item,
+    variants: &IndexVec<VariantIdx, clean::Item>,
+) -> String {
+    let tcx = cx.tcx();
+    let cache = cx.cache();
+    let v: Vec<&Impl> = cx.shared.all_impls_for_item(it, it.item_id.expect_def_id());
+
+    let default_fqp = vec![sym::core, kw::Default, sym::Default];
+    let mut label = "".to_string();
+    v.into_iter().for_each(|i| {
+        if let clean::ImplItem(ref impl_) = *i.impl_item.kind && let Some(trait_) = &impl_.trait_ {
+            let did = trait_.def_id();
+
+            let fqp = if let Some(&(ref fqp, _)) = cache.paths.get(&did) {
+                Some(fqp)
+            } else {
+                if let Some(&(ref fqp, _)) = cache.external_paths.get(&did) {
+                    Some(fqp)
+                } else {
+                    None
+                }
+            };
+
+            if let Some(fqp) = fqp {
+                if fqp == &default_fqp {
+                    let has_default_on_variant = variants.iter().any(|variant| {
+                        variant.attrs.other_attrs.iter().any(|a| a.has_name(kw::Default))
+                    });
+
+                    let id = get_id_for_impl(tcx, i.impl_item.item_id);
+                    if !has_default_on_variant {
+                        label = chip(&format!("<a href=\"#{}\">Default</a>", id), "default");
```

Comment:
If you turn the `for_each` into a `for` loop you could just return early here and get rid of the variable `label` altogether, right?

```suggestion
                        return chip(&format!("<a href=\"#{id}\">Default</a>"), "default");
```

Reply:
done

Reply:
done